Makefile

```diff
@@ -7,8 +7,8 @@ clean:
 
 simpletests: test-src/*.c test-src/*.cpp
 	mkdir -p test-bin
-	for i in test-src/*.c test-src/*.cpp; do $(CC) $(CFLAGS) -o test-bin/`basename $$i .c` $$i -lpthread; done
+	for i in test-src/*.c test-src/*.cpp; do $(CC) -lpthread $(CFLAGS) -o test-bin/`basename $$i .c` $$i; done
 
 complicatedtests: test-src/*/*.c test-src/*.cpp
 	mkdir -p test-bin
-	for i in test-src/*/; do pushd $$i; make -B; popd; cp $$i/`basename $$i` test-bin/`basename $$i`; done
+	for i in test-src/*/; do pushd $$i; gmake -B; popd; cp $$i/`basename $$i` test-bin/`basename $$i`; done
```

Comment:
The symlink from gmake to make doesn't exist on modern Debian/Ubuntu so this will break; I can see why you need to specify it though